### ISSUE TITLE
CUDA Out of Memory: Convert 3DGS to Mesh with NeRF and Marching Cubes

### ISSUE THREAD

**Issue by tcarneyjohnson** [MrForExample/ComfyUI-3D-Pack#74]
First of all, let me just say a huge **thank you** for all your hard work on this.

Right now I'm receiving a CUDA Out of Memory error when I try to run the Convert 3DGS to Mesh with NeRF and Marching Cubes node in the MVDream_to_Large_Multiview_Gaussian_Model_to_3DGS and_3DMesh workflow. (The error message is added to the bottom of this message).

Some additional information:
-Windows 10 machine
-Titan RTX video card (seemed like a good idea at the time)
-----video card drivers are up to date (using studio version)
-VS Build Tools for 2022/2019 installed, including Desktop development with C++.
-CUDA 12.1 is set for CUDA_PATH in my environmental variables
-Trying to use Install method 0 by installing directly into the Comfy3D environment without Miniconda
-Have tried to purge cache and reinstall scikit-learn, other dependencies, etc. in addition to adding --force-reinstall command to _install_windows_portable_win_py311_cu121.bat_ file
-I have also added C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Scripts to path
-Most of the other workflows seem to function without error.
-----I also get an error about not having a kernel loaded when I try to run the Triplane Gaussian workflow. I am not sure if that's related at this point and I can provide more information if it seems relevant. (Right now I'm trying to take things one issue at a time.)

Thank you in advance for any help or insight you can offer!

_The error shown in the ComfyUI window is as follows:_

Error occurred when executing [Comfy3D] Convert 3DGS to Mesh with NeRF and Marching Cubes:

Allocation on device 0 would exceed allowed memory. (out of memory)
Currently allocated : 4.63 GiB
Requested : 277.99 GiB
Device limit : 24.00 GiB
Free (according to CUDA): 17.59 GiB
PyTorch limit (set by user-supplied memory fraction)
: 17179869184.00 GiB

File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\execution.py", line 152, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\execution.py", line 82, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\execution.py", line 75, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\nodes.py", line 1454, in convert_gs_ply
converter.fit_nerf()
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\algorithms\nerf_marching_cubes_converter.py", line 174, in fit_nerf
image_gt, alpha_gt = self.render_gs(pose)
^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\algorithms\nerf_marching_cubes_converter.py", line 80, in render_gs
out = self.gs_renderer.render(self.gaussians.unsqueeze(0), cam_view.unsqueeze(0), cam_view_proj.unsqueeze(0), cam_pos.unsqueeze(0))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\lgm\core\gs.py", line 76, in render
rendered_image, radii, rendered_depth, rendered_alpha = rasterizer(
^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1511, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1520, in _call_impl
return forward_call(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\diff_gaussian_rasterization\__init__.py", line 213, in forward
return rasterize_gaussians(
^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\diff_gaussian_rasterization\__init__.py", line 32, in rasterize_gaussians
return _RasterizeGaussians.apply(
^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\autograd\function.py", line 553, in apply
return super().apply(*args, **kwargs) # type: ignore[misc]
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\Users\Johnson\Desktop\ComfyUI_windows_portable\python_embeded\Lib\site-packages\diff_gaussian_rasterization\__init__.py", line 92, in forward
num_rendered, color, depth, alpha, radii, geomBuffer, binningBuffer, imgBuffer = _C.rasterize_gaussians(*args)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

**Comment by tcarneyjohnson** [comment]
Thank you anyway, you've been more helpful than you realize!

After a bit of reading in the other threads, I suspect that Pytorch needs to be built from source for my specific machine/GPU, which would also explain the kernel error in the Triplane Gaussian workflow.

(Trying to find the link to the other forum I read, but from what I understand Just because Pytorch says CUDA is available, doesn't mean it's properly configured for your machine.)

I will let you know how it goes.

**Comment by merecesarchviz** [comment]
This is frustrating! after solve so many issues during install, no also in comfyui i got this error! how we can solve? thks

![imagem](https://github.com/MrForExample/ComfyUI-3D-Pack/assets/79140456/a5ecc402-61df-4ddc-950a-fab12e011509)

**Comment by MrForExample** [comment]
@merecesarchviz 
Hi, what kind of install method that you are using? And can you show me the full error log pls?

**Comment by merecesarchviz** [comment]
I use the Install Method 0: Directly inside ComfyUI Python Embeded Environment

here is the full error:

Loading pipeline components...: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:01<00:00,  4.32it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:04<00:00,  6.73it/s]
[Comfy3D] [Load_Large_Multiview_Gaussian_Model] loaded model ckpt from F:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\checkpoints/lgm\model_fp16.safetensors
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
ERROR:root:!!! Exception during processing !!!
ERROR:root:Traceback (most recent call last):
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 2096, in _run_ninja_build
    subprocess.run(
  File "subprocess.py", line 571, in run
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "F:\ComfyUI_windows_portable\ComfyUI\execution.py", line 152, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "F:\ComfyUI_windows_portable\ComfyUI\execution.py", line 82, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "F:\ComfyUI_windows_portable\ComfyUI\execution.py", line 75, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "F:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\nodes.py", line 1453, in convert_gs_ply
    converter = GSConverterNeRFMarchingCubes(config_defaults[gs_config], gs_ply).cuda()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "F:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\algorithms\nerf_marching_cubes_converter.py", line 27, in __init__
    from kiui.gridencoder import GridEncoder
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\__init__.py", line 1, in <module>
    from .grid import GridEncoder
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\grid.py", line 9, in <module>
    from .backend import _backend
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\backend.py", line 48, in <module>
    _backend = load(
               ^^^^^
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 1306, in load
    return _jit_compile(
           ^^^^^^^^^^^^^
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 1710, in _jit_compile
    _write_ninja_file_and_build_library(
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 1823, in _write_ninja_file_and_build_library
    _run_ninja_build(
  File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 2112, in _run_ninja_build
    raise RuntimeError(message) from e
RuntimeError: Error building extension '_grid_encoder': [1/2] C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc --generate-dependencies-with-compile --dependency-output gridencoder.cuda.o.d -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4068 -Xcompiler /wd4067 -Xcompiler /wd4624 -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -DTORCH_EXTENSION_NAME=_grid_encoder -DTORCH_API_INCLUDE_EXTENSION_H -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\torch\csrc\api\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\TH -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -IF:\ComfyUI_windows_portable\python_embeded\Include -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_86,code=sm_86 -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -std=c++17 -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -c F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\src\gridencoder.cu -o gridencoder.cuda.o
FAILED: gridencoder.cuda.o
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc --generate-dependencies-with-compile --dependency-output gridencoder.cuda.o.d -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4068 -Xcompiler /wd4067 -Xcompiler /wd4624 -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -DTORCH_EXTENSION_NAME=_grid_encoder -DTORCH_API_INCLUDE_EXTENSION_H -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\torch\csrc\api\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\TH -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -IF:\ComfyUI_windows_portable\python_embeded\Include -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_86,code=sm_86 -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -std=c++17 -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -c F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\src\gridencoder.cu -o gridencoder.cuda.o
nvcc fatal   : Unsupported gpu architecture 'compute_89'
ninja: build stopped: subcommand failed.


Prompt executed in 45.90 seconds

**Comment by merecesarchviz** [comment]
> @merecesarchviz Hi, what kind of install method that you are using? And can you show me the full error log pls?

I use the Install Method 0: Directly inside ComfyUI Python Embeded Environment

here is the full error:

Loading pipeline components...: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:01<00:00, 4.32it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:04<00:00, 6.73it/s]
[Comfy3D] [Load_Large_Multiview_Gaussian_Model] loaded model ckpt from F:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\checkpoints/lgm\model_fp16.safetensors
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
ERROR:root:!!! Exception during processing !!!
ERROR:root:Traceback (most recent call last):
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 2096, in _run_ninja_build
subprocess.run(
File "subprocess.py", line 571, in run
subprocess.CalledProcessError: Command '['ninja', '-v']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "F:\ComfyUI_windows_portable\ComfyUI\execution.py", line 152, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "F:\ComfyUI_windows_portable\ComfyUI\execution.py", line 82, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "F:\ComfyUI_windows_portable\ComfyUI\execution.py", line 75, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "F:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\nodes.py", line 1453, in convert_gs_ply
converter = GSConverterNeRFMarchingCubes(config_defaults[gs_config], gs_ply).cuda()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "F:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-3D-Pack\algorithms\nerf_marching_cubes_converter.py", line 27, in init
from kiui.gridencoder import GridEncoder
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder_init_.py", line 1, in
from .grid import GridEncoder
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\grid.py", line 9, in
from .backend import backend
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\backend.py", line 48, in
backend = load(
^^^^^
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 1306, in load
return jit_compile(
^^^^^^^^^^^^^
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 1710, in jit_compile
write_ninja_file_and_build_library(
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 1823, in write_ninja_file_and_build_library
run_ninja_build(
File "F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\utils\cpp_extension.py", line 2112, in run_ninja_build
raise RuntimeError(message) from e
RuntimeError: Error building extension 'grid_encoder': [1/2] C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc --generate-dependencies-with-compile --dependency-output gridencoder.cuda.o.d -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4068 -Xcompiler /wd4067 -Xcompiler /wd4624 -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -DTORCH_EXTENSION_NAME=grid_encoder -DTORCH_API_INCLUDE_EXTENSION_H -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\torch\csrc\api\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\TH -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -IF:\ComfyUI_windows_portable\python_embeded\Include -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS -D__CUDA_NO_HALF_CONVERSIONS -D__CUDA_NO_BFLOAT16_CONVERSIONS -D__CUDA_NO_HALF2_OPERATORS --expt-relaxed-constexpr -gencode=arch=compute_86,code=sm_86 -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -std=c++17 -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -c F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\src\gridencoder.cu -o gridencoder.cuda.o
FAILED: gridencoder.cuda.o
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\bin\nvcc --generate-dependencies-with-compile --dependency-output gridencoder.cuda.o.d -Xcudafe --diag_suppress=dll_interface_conflict_dllexport_assumed -Xcudafe --diag_suppress=dll_interface_conflict_none_assumed -Xcudafe --diag_suppress=field_without_dll_interface -Xcudafe --diag_suppress=base_class_has_different_dll_interface -Xcompiler /EHsc -Xcompiler /wd4068 -Xcompiler /wd4067 -Xcompiler /wd4624 -Xcompiler /wd4190 -Xcompiler /wd4018 -Xcompiler /wd4275 -Xcompiler /wd4267 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4819 -Xcompiler /MD -DTORCH_EXTENSION_NAME=grid_encoder -DTORCH_API_INCLUDE_EXTENSION_H -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\torch\csrc\api\include -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\TH -IF:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\include\THC "-IC:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.7\include" -IF:\ComfyUI_windows_portable\python_embeded\Include -D_GLIBCXX_USE_CXX11_ABI=0 -D__CUDA_NO_HALF_OPERATORS_ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr -gencode=arch=compute_86,code=sm_86 -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -std=c++17 -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -gencode=arch=compute_89,code=compute_89 -gencode=arch=compute_89,code=sm_89 -c F:\ComfyUI_windows_portable\python_embeded\Lib\site-packages\kiui\gridencoder\src\gridencoder.cu -o gridencoder.cuda.o
nvcc fatal : Unsupported gpu architecture 'compute_89'
ninja: build stopped: subcommand failed.

Prompt executed in 45.90 seconds

**Comment by tcarneyjohnson** [comment]
@merecesarchviz

Is your CUDA_PATH environmental variable set to version 12.x? I think I ran into the same error while tinkering and setting it to 11.8.

**Comment by tcarneyjohnson** [comment]
@merecesarchviz
I think CUDA_PATH needs to be set to at least version 12, but I am not sure.

**Comment by tcarneyjohnson** [comment]
As an update to my CUDA OOM error:

I have built Pytorch from source and it **seems** to check out in my Miniconda environment, but I do not know how to get it working in my main Windows environment or ComfyUI.

I have tried installing it via navigating to my compiled pytorch directory and running
"pip install ." from cmd. I have also tried "python setup.py install" with no luck. I will add the error message I get below while trying to confirm it is working.

Any thoughts or suggestions would be greatly appreciated.

C:\Users\Johnson\pytorch>pip install --force-reinstall .
Processing c:\users\johnson\pytorch
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting filelock (from torch==2.3.0a0+git12191f4)
  Using cached filelock-3.13.1-py3-none-any.whl.metadata (2.8 kB)
Collecting typing-extensions>=4.8.0 (from torch==2.3.0a0+git12191f4)
  Using cached typing_extensions-4.10.0-py3-none-any.whl.metadata (3.0 kB)
Collecting sympy (from torch==2.3.0a0+git12191f4)
  Using cached sympy-1.12-py3-none-any.whl.metadata (12 kB)
Collecting networkx (from torch==2.3.0a0+git12191f4)
  Using cached networkx-3.2.1-py3-none-any.whl.metadata (5.2 kB)
Collecting jinja2 (from torch==2.3.0a0+git12191f4)
  Using cached Jinja2-3.1.3-py3-none-any.whl.metadata (3.3 kB)
Collecting fsspec (from torch==2.3.0a0+git12191f4)
  Using cached fsspec-2024.2.0-py3-none-any.whl.metadata (6.8 kB)
Collecting mkl<=2021.4.0,>=2021.1.1 (from torch==2.3.0a0+git12191f4)
  Using cached mkl-2021.4.0-py2.py3-none-win_amd64.whl.metadata (1.4 kB)
Collecting intel-openmp==2021.* (from mkl<=2021.4.0,>=2021.1.1->torch==2.3.0a0+git12191f4)
  Using cached intel_openmp-2021.4.0-py2.py3-none-win_amd64.whl.metadata (1.2 kB)
Collecting tbb==2021.* (from mkl<=2021.4.0,>=2021.1.1->torch==2.3.0a0+git12191f4)
  Using cached tbb-2021.11.0-py3-none-win_amd64.whl.metadata (1.1 kB)
Collecting MarkupSafe>=2.0 (from jinja2->torch==2.3.0a0+git12191f4)
  Using cached MarkupSafe-2.1.5-cp312-cp312-win_amd64.whl.metadata (3.1 kB)
Collecting mpmath>=0.19 (from sympy->torch==2.3.0a0+git12191f4)
  Using cached mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Using cached mkl-2021.4.0-py2.py3-none-win_amd64.whl (228.5 MB)
Using cached intel_openmp-2021.4.0-py2.py3-none-win_amd64.whl (3.5 MB)
Using cached tbb-2021.11.0-py3-none-win_amd64.whl (298 kB)
Using cached typing_extensions-4.10.0-py3-none-any.whl (33 kB)
Using cached filelock-3.13.1-py3-none-any.whl (11 kB)
Using cached fsspec-2024.2.0-py3-none-any.whl (170 kB)
Using cached Jinja2-3.1.3-py3-none-any.whl (133 kB)
Using cached networkx-3.2.1-py3-none-any.whl (1.6 MB)
Using cached sympy-1.12-py3-none-any.whl (5.7 MB)
Using cached MarkupSafe-2.1.5-cp312-cp312-win_amd64.whl (17 kB)
Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Building wheels for collected packages: torch
  Building wheel for torch (pyproject.toml) ... done
  Created wheel for torch: filename=torch-2.3.0a0+git12191f4-cp312-cp312-win_amd64.whl size=251399679 sha256=390e9f73d5c6b05403770cc4025a52d73be89e9632f2a012de803715df9e1f2e
  Stored in directory: C:\Users\Johnson\AppData\Local\Temp\pip-ephem-wheel-cache-tjikoaja\wheels\dc\e8\21\16b47c0beab9d071c32844983a61318b7c09727a397a67a755
Successfully built torch
Installing collected packages: tbb, mpmath, intel-openmp, typing-extensions, sympy, networkx, mkl, MarkupSafe, fsspec, filelock, jinja2, torch
  Attempting uninstall: tbb
    Found existing installation: tbb 2021.11.0
    Uninstalling tbb-2021.11.0:
      Successfully uninstalled tbb-2021.11.0
  Attempting uninstall: mpmath
    Found existing installation: mpmath 1.3.0
    Uninstalling mpmath-1.3.0:
      Successfully uninstalled mpmath-1.3.0
  Attempting uninstall: intel-openmp
    Found existing installation: intel-openmp 2021.4.0
    Uninstalling intel-openmp-2021.4.0:
      Successfully uninstalled intel-openmp-2021.4.0
  Attempting uninstall: typing-extensions
    Found existing installation: typing_extensions 4.10.0
    Uninstalling typing_extensions-4.10.0:
      Successfully uninstalled typing_extensions-4.10.0
  Attempting uninstall: sympy
    Found existing installation: sympy 1.12
    Uninstalling sympy-1.12:
      Successfully uninstalled sympy-1.12
  Attempting uninstall: networkx
    Found existing installation: networkx 3.2.1
    Uninstalling networkx-3.2.1:
      Successfully uninstalled networkx-3.2.1
  Attempting uninstall: mkl
    Found existing installation: mkl 2021.4.0
    Uninstalling mkl-2021.4.0:
      Successfully uninstalled mkl-2021.4.0
  Attempting uninstall: MarkupSafe
    Found existing installation: MarkupSafe 2.1.5
    Uninstalling MarkupSafe-2.1.5:
      Successfully uninstalled MarkupSafe-2.1.5
  Attempting uninstall: fsspec
    Found existing installation: fsspec 2024.2.0
    Uninstalling fsspec-2024.2.0:
      Successfully uninstalled fsspec-2024.2.0
  Attempting uninstall: filelock
    Found existing installation: filelock 3.13.1
    Uninstalling filelock-3.13.1:
      Successfully uninstalled filelock-3.13.1
  Attempting uninstall: jinja2
    Found existing installation: Jinja2 3.1.3
    Uninstalling Jinja2-3.1.3:
      Successfully uninstalled Jinja2-3.1.3
Successfully installed MarkupSafe-2.1.5 filelock-3.13.1 fsspec-2024.2.0 intel-openmp-2021.4.0 jinja2-3.1.3 mkl-2021.4.0 mpmath-1.3.0 networkx-3.2.1 sympy-1.12 tbb-2021.11.0 torch-2.3.0a0+git12191f4 typing-extensions-4.10.0

C:\Users\Johnson\pytorch>python
Python 3.12.2 (tags/v3.12.2:6abddd9, Feb  6 2024, 21:26:36) [MSC v.1937 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Johnson\pytorch\torch\__init__.py", line 141, in <module>
    raise err
OSError: [WinError 126] The specified module could not be found. Error loading "C:\Users\Johnson\pytorch\torch\lib\backend_with_compiler.dll" or one of its dependencies.
>>>

**Comment by tcarneyjohnson** [comment]
So, as a final update on this, I never managed to get this installed correctly in my main Windows 10 environment, but it works flawlessly via the recommended Miniconda environment.

Final thoughts on a possible explanations for the CUDA OOM error from a non-programmer:

- I've read that conda and pip pull from different repositories, so it's possible some small variation in what's being downloaded and installed is causing the error. It could also be that there's something being installed in the Miniconda environment via the .bat file that's missing for the main Windows environment.

- It seems like this error has to do with cuda-malloc and pytorch memory settings combined, although the --disable-cuda-malloc setting did not resolve it. It looks like pytorch wants a variable for set_per_process_memory_fraction, which isn't being set correctly in the main environment. I have no idea how or where to set this.

- **Bottom line: If you're running into the Out of Memory Error, just use the Miniconda install.**

Anyone who is struggling with a "Missing Nodes" error after following the installation steps should **_make sure that they have the RemBG node extension installed_**  (along with whatever other specific extension the workflow lists) as that finally made the 3D nodes available for me. This error is a little less obvious because it's not listed as a missing node, and shows up as an error in the console. I'm pretty sure this was my initial sticking point with the Miniconda environment install. 

@MrForExample 
Thanks again for your time and this awesome work, and most of all KEEP GOING!!!
